src/error.F90

```diff
@@ -18,24 +18,36 @@ module error
 ! stream.
 !===============================================================================
 
-  subroutine warning(message)
+  subroutine warning(message_in, deprecation)
 
-    character(*) :: message
-
-    integer :: i_start   ! starting position
-    integer :: i_end     ! ending position
-    integer :: line_wrap ! length of line
-    integer :: length    ! length of message
-    integer :: indent    ! length of indentation
+    character(*)                  :: message_in  ! printed with the warning
+    logical, optional, intent(in) :: deprecation ! is a depreciation warning
 
-    ! Write warning at beginning
-    write(ERROR_UNIT, fmt='(1X,A)', advance='no') 'WARNING: '
+    integer                   :: i_start      ! starting position
+    integer                   :: i_end        ! ending position
+    integer                   :: line_wrap    ! length of line
+    integer                   :: length       ! length of message
+    integer                   :: indent       ! length of indentation
+    character(:), allocatable :: message      ! input message with a prefix
```

Comment:
Unfortunately, allocatable characters break building with gfortran 4.6. Normally, I wouldn't care much, but since we're moving to Travis CI for testing and their VMs still use Ubuntu 12.04 (which in turn uses gcc 4.6), this would prevent building there. Would you mind re-coding this to remove allocatable characters?